gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/OrchestratorTest.java

```diff
@@ -244,9 +250,10 @@ public void createTopologySpec() {
     }
     // Make sure TopologyCatalog is empty
     Assert.assertTrue(specs.size() == 0, "Spec store should be empty before addition");
-    // Make sure TopologyCatalog Listener is empty
-    Assert.assertTrue(specCompiler.getTopologySpecMap().size() == 0, "SpecCompiler should not know about any Topology "
-        + "before addition");
+    Assert.assertTrue(specCompiler.getTopologySpecMap().size() == 1, "SpecCompiler should know about any Topology "
```

Comment:
desc here makes it sound like the test should be `> 1`.  to that end, I didn't notice any `Preconditions` check or similar to insist on a non-empty collection.  do we want one?

Reply:
"SpecCompiler should know about any Topology irrespective of what is there in the topology catalog" ? how does that sound like test should be `>1` ? What description do you suggest?

There is no precondition. SpecCompiler should just know about the topologies. Number of topologies known to spec compiler should just always be 1 (equals to total number of topologies).

Reply:
"know about *any*" suggested it would apply even if there were > 1 topo specs.  couldn't there be more than one?

also, if non-empty `Set<TopologySpec>` is critical, I was suggesting a guava `Precondition`